summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala

```diff
@@ -144,8 +142,35 @@ private[storm] case class StormTopologyBuilder(options: Map[String, Options], jo
     case Some(props) =>
       registerShardedKeyValue[K, V](topology, node, provider, props.batcher, props.shards)
     case None =>
-      throw new IllegalStateException("Shouldn't happen, trying to register component which emits " +
-        "keyed values and don't have downstream summers.")
+      registerKeyValue[K, V](topology, node, provider)
```

Comment:
This method represents high level strategy how to deal with key value components, all further logic extracted into `registerAggregatedEdges`, `registerShardedKeyValue` and `registerKeyValue`. I can extract `aggregated` part further but I prefer to keep it as is. Also I doubt it will grow more.